lib/app/models/testing_instance.rb

```diff
@@ -281,6 +275,40 @@ def token_expiration_time
         token_retrieved_at + token_expires_in.seconds
       end
 
+      def get_sequence_requirement(requirement)
+        return unless requirement&.dig(:name)
+
+        attributes = { name: requirement[:name],
+                       testing_instance: self,
+                       label: requirement[:name],
+                       value: '' }
+          .merge(requirement)
+
+        condition_attributes = [:name, :testing_instance]
+
+        # Get the conditions that determine whether the record already exists
+        conditions = attributes.select { |key, _value| condition_attributes.include?(key) }
+
+        SequenceRequirement.first_or_create(conditions, attributes)
+      end
+
+      def add_sequence_requirements(requirements)
+        return unless requirements.present?
+
+        requirements.each do |requirement, texts|
+          get_sequence_requirement(name: requirement.to_s, label: texts[:label], description: texts[:description])
+        end
+      end
+
+      def get_requirement_value(requirement_name)
+        get_sequence_requirement(name: requirement_name.to_s, testing_instance: self).value
+      end
+
+      def set_requirement_value(requirement_name, value)
+        get_sequence_requirement(name: requirement_name.to_s, testing_instance: self).update(value: value)
+        self.patient_id = value if requirement_name == 'patient_id'
```

Comment:
I hope we're not keeping this in the final version.

Reply:
Took this out. Forgot we have the setters now so I'm back to using that in the test_set_endpoint